### PR TITLE
Implement release process

### PR DESCRIPTION
**Pull request by strieflin**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,10 +1,18 @@
 #
-# Copyright (c) 2021 - for information on the respective copyright owner
+# Copyright (c) 2021-2023 - for information on the respective copyright owner
 # see the NOTICE file and/or the repository https://github.com/carbynestack/castor.
```
Comment (sbckr) [inline .pre-commit-config.yaml:3]: wrong repository?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+  - repo: https://github.com/alessandrojcm/commitlint-pre-commit-hook
+    rev: v9.5.0
+    hooks:
+      - id: commitlint
+        stages:
+          - commit-msg
+        additional_dependencies:
+          - "@commitlint/config-conventional"
   - repo: https://github.com/executablebooks/mdformat
     rev: 0.7.7
```
Comment (sbckr) [inline .pre-commit-config.yaml:17]: we used 0.7.16 for the other repos

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -18,4 +26,4 @@ repos:
     rev: v0.27.1
```
Comment (sbckr) [inline .pre-commit-config.yaml:26]: we upgrade other repos to v0.33.0

File: release-please-config.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "bump-minor-pre-major": true,
+    "separate-pull-requests": true,
+    "pull-request-title-pattern": "chore: release ${component} ${version}",
+    "pull-request-header": ":package: Staging a new release",
+    "packages": {
+        ".": {
+            "package-name": "sdk",
```
Comment (sbckr) [inline release-please-config.json:8]: Not sure about **_sdk_** as package name, but can't come up with any better

**Review by sbckr — CHANGES_REQUESTED** [review]
see individual comments

File: release-please-config.json (hunk 4 above)
Comment (strieflin) [reply release-please-config.json:8]: Well if have nothing else, that is what we used before.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (strieflin) [reply .pre-commit-config.yaml]: Fixed

File: .pre-commit-config.yaml (hunk 2 above)
Comment (strieflin) [reply .pre-commit-config.yaml]: Fixed

File: .pre-commit-config.yaml (hunk 3 above)
Comment (strieflin) [reply .pre-commit-config.yaml]: Fixed

**Review by sbckr — APPROVED** [review]
LGTM

**Pull request merged**